tests/unit/infrastructure/s3/s3ZipService.test.ts

```diff
@@ -0,0 +1,107 @@
+import { S3 } from 'aws-sdk';
+import getFromS3 from '../../../../src/infrastructure/s3/s3ZipService';
+
+describe('S3 Zip Service', () => {
+  afterEach(() => {
+    jest.resetAllMocks().restoreAllMocks();
+  });
+
+  it('passes the expected key to getObject if folder is defined', async () => {
+    const mockS3 = ({} as unknown) as S3;
+    const bucket = 'bucket';
+    const folder = 'folder';
+    const adrDocumentId = '1234';
+    const mockPromise = jest
+      .fn()
+      .mockResolvedValue({ Body: 'Success!', ContentType: 'application/octet-stream' });
+    const mockGetObject = jest.fn().mockReturnValue({ promise: mockPromise });
+
+    mockS3.getObject = mockGetObject;
+    await getFromS3(mockS3, bucket, folder, adrDocumentId).catch(() => {});
+
+    const firstCall = mockGetObject.mock.calls[0] as S3.GetObjectRequest[];
+    const firstArg = firstCall[0];
+
+    expect(firstArg.Key).toBe(`${folder}/adr-documents/${adrDocumentId}.zip`);
+  });
+
+  it('passes the expected key to getObject if folder is undefined', async () => {
+    const mockS3 = ({} as unknown) as S3;
+    const bucket = 'bucket';
+    const folder = undefined;
+    const adrDocumentId = '1234';
+    const mockPromise = jest
+      .fn()
+      .mockReturnValue(Promise.resolve({ Body: 'Success!', ContentType: 'application/octet-stream' }));
+    const mockGetObject = jest.fn().mockReturnValue({ promise: mockPromise });
+
+    mockS3.getObject = mockGetObject;
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
```

Comment:
what are you switching off the eslint?

Reply:
Even though the method can take a string | undefined, as I defined the folder as undefined it was defaulting to type of any.

Have pushed up change to definition of folder and remove the eslint line.